### PR TITLE
Test/fix scoping rules for pattern locals declared within a fixed statement.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Related to #8817.

@gafter, @agocke, @dotnet/roslyn-compiler Please review.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@gafter, @agocke, @dotnet/roslyn-compiler Please review.

File: src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs [hunk 1]
```diff
@@ -1509,16 +1509,20 @@ public override BoundNode VisitUsingStatement(BoundUsingStatement node)
...
             {
-                if (local.DeclarationKind == LocalDeclarationKind.RegularVariable)
-                {
-                    ReportIfUnused(local, assigned: true);
-                }
-                else
+                switch (local.DeclarationKind)
                 {
-                    NoteRead(local); // At the end of the statement, there's an implied read when the local is disposed
+                    case LocalDeclarationKind.RegularVariable:
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs]: Why not for pattern variables?

File: src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs [hunk 2]
```diff
@@ -1509,16 +1509,20 @@ public override BoundNode VisitUsingStatement(BoundUsingStatement node)
...
+                        break;
+
+                    case LocalDeclarationKind.UsingVariable:
+                        NoteRead(local); // At the end of the statement, there's an implied read when the local is disposed
+                        break;
                 }
             }
-            Debug.Assert(localsOpt.All(_usedVariables.Contains));
+
+            // PROTOTYPE: Should the filter for pattern variables stay?
```
Comment (gafter) [inline src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs]: Please make the comment prefix `// PROTOTYPE(patterns):`

File: src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs (hunk 1 above)
Comment (gafter) [inline src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs]: Never mind, I know why: sometimes the syntax requires you to declare a variable you have no intention to use.

**Comment by gafter** [comment]
LGTM

File: src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs (hunk 2 above)
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs]: I'll make this change in the next PR

**Comment by AlekseyTs** [comment]
@agocke, @dotnet/roslyn-compiler Please review, I need a second sign-off for a small product change.

File: src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs (hunk 1 above)
Comment (agocke) [inline src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs]: @gafter Could you elaborate? I'm not clear on the case you're describing.

**Comment by agocke** [comment]
:+1:

**Pull request merged**